aws/resource_aws_dx_connection.go

```diff
@@ -0,0 +1,116 @@
+package aws
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/directconnect"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsDxConnection() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsDxConnectionCreate,
+		Read:   resourceAwsDxConnectionRead,
+		Delete: resourceAwsDxConnectionDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"band_width": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateDxConnectionBandWidth,
+			},
+			"location": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceAwsDxConnectionCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dxconn
+
+	input := &directconnect.CreateConnectionInput{
+		Bandwidth:      aws.String(d.Get("band_width").(string)),
+		ConnectionName: aws.String(d.Get("name").(string)),
+		Location:       aws.String(d.Get("location").(string)),
+	}
+	resp, err := conn.CreateConnection(input)
+	if err != nil {
+		return err
+	}
+	d.SetId(*resp.ConnectionId)
+	return resourceAwsDxConnectionRead(d, meta)
+}
+
+func resourceAwsDxConnectionRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dxconn
+
+	connectionId := d.Id()
+	input := &directconnect.DescribeConnectionsInput{
+		ConnectionId: aws.String(connectionId),
+	}
+	resp, err := conn.DescribeConnections(input)
+	if err != nil {
+		d.SetId("")
+		return err
+	}
+	if len(resp.Connections) != 1 {
+		d.SetId("")
+		return fmt.Errorf("[ERROR] Number of DX Connection (%s) isn't one, got %d", connectionId, len(resp.Connections))
+	}
+	if d.Id() != *resp.Connections[0].ConnectionId {
+		d.SetId("")
+		return fmt.Errorf("[ERROR] DX Connection (%s) not found", connectionId)
+	}
+	return nil
+}
+
+func resourceAwsDxConnectionDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dxconn
+
+	input := &directconnect.DeleteConnectionInput{
+		ConnectionId: aws.String(d.Id()),
+	}
+	_, err := conn.DeleteConnection(input)
+	if err != nil {
+		return err
+	}
+	deleteStateConf := &resource.StateChangeConf{
+		Pending:    []string{directconnect.ConnectionStatePending, directconnect.ConnectionStateOrdering, directconnect.ConnectionStateAvailable, directconnect.ConnectionStateRequested},
+		Target:     []string{directconnect.ConnectionStateDeleted, directconnect.ConnectionStateDeleting},
```

Comment:
Any particular reason to treat `directconnect.ConnectionStateDeleting` as target here?